lib/internal/repl.js

```diff
@@ -84,18 +90,26 @@ function setupHistory(repl, historyPath, ready) {
     if (err) {
       return ready(err);
     }
-
-    if (data) {
+    if (data && data[0] === '['
+        // Check the last two characters in case there is a trailing newline
+        && /\]/.test(data.substr(-2))) {
```

Comment:
Alternatively, maybe prepend the plain text version with a magic NUL as the first character?